ophyd/utils/epics_pvs.py

```diff
@@ -325,3 +326,13 @@ def records_from_db(fn):
         ret.append((rtype, record))
 
     return ret
+
+def raise_if_disconnected(fcn):
+    '''Decorator to catch attempted access to disconnected EPICS channels.'''
+    @functools.wraps(fcn)
+    def wrapper(self, *args, **kwargs):
+        if self.connected:
+            return fcn(self, *args, **kwargs)
+        else:
+            raise Exception('{} is not connected'.format(self.name))
```

Comment:
Anything's better than `Exception` - as the caller, you have no idea what happened if something so generic is raised.

Personally, I like when packages which define their own exceptions use a common base class. That's what I was going for in `ophyd/utils/errors.py`.